.github/workflows/presubmit.yml

```diff
@@ -0,0 +1,36 @@
+name: SmoothApp Presubmit Tests
+
+on: pull_request
+    
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@v1
+
+      # Setup flutter 
+      - name: "Setup Flutter"
+        uses: subosito/flutter-action@v1
+        with:
+          channel: 'stable' 
+          # flutter-version: '1.12.x'
+      
+      # Get dependencies.
+      - name: "Get dependencies"
+        run: ci/pub_upgrade.sh
+      
+      # Check for formatting issues
+      - name: Check for formatting issues (run "flutter format . ")
+        run: flutter format --set-exit-if-changed .
```

Comment:
We now let both workflows fail when there are formatting errors, I am correct in thinking that we then do not need an autoformat workflow

Reply:
I agree, we probably don't need to do auto-formatting in a workflow. It seems better to me to let the developer run the formatting command (either in their IDE, or manually), since it's possible to create merge conflicts when you auto-format, and because you get to see the results of the formatting locally and don't need to pull to get the automatic changes. Sometimes you want to play with the code to get better formatting too.